src/c1_state_machine/p3_atm.rs

```diff
@@ -82,40 +81,164 @@ fn sm_3_simple_swipe_card() {
 
 #[test]
 fn sm_3_swipe_card_again_part_way_through() {
-    todo!()
+    let start = Atm {
+        cash_inside: 10,
+        expected_pin_hash: Auth::Authenticating(1234),
+        keystroke_register: Vec::new(),
+    };
+    let end = Atm::next_state(&start, &Action::SwipeCard(1234));
+    let expected = Atm {
+        cash_inside: 10,
+        expected_pin_hash: Auth::Authenticating(1234),
+        keystroke_register: Vec::new(),
+    };
+
+    assert_eq!(end, expected);
```

Comment:
This is a good test. We should leave it exactly as it is. We should also add another test for is the user has entered a few digits of the pin and then swipes the card again.

I can imagine two reasonable post-states:
1. Nothing happens. The state is the same as before with the old expected pin and the few digits keyed in
2. It starts a brand new session with the new (potentially different) expected pin and no digits keyed in.

I don't have an opinion on which is better. Whichever you prefer.

Reply:
I'm taking it as that the machine won't even accept the card so it remains in the same state ;)